lib/AST/Decl.cpp

```diff
@@ -493,6 +493,29 @@ unsigned Decl::getAttachedMacroDiscriminator(DeclBaseName macroName,
       macroName);
 }
 
+Type Decl::getResolvedCustomAttrType(CustomAttr *attr) const {
```

Comment:
What are the other places that evaluate these requests directly? Should they use the new method too? Otherwise it's a bit odd that only serialization calls it

Reply:
I've got one follow up planned to adopt this in interface printing. I audited all the call sites and in all but the serialization and printing cases we have already run a request to get the `CustomAttr` corresponding to the global actor/property wrapper/result builder, etc. Those requests cause the type to be resolved themselves, so going through this interface in those spots would be unnecessary, though it also wouldn't be harmful I suppose. It seems Serialization and Printing seem to be the only two places where we operate on a `CustomAttr` abstractly.